frontend/packages/gitops-plugin/src/components/utils/useEnvDetails.ts

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+import { fetchAppGroups } from './gitops-utils';
+import { GitOpsAppGroupData } from './gitops-types';
+
+const useEnvDetails = (appName, manifestURL, pipelinesBaseURI) => {
+  const { t } = useTranslation();
+  const [envs, setEnvs] = React.useState<string[]>(null);
+  const [emptyStateMsg, setEmptyStateMsg] = React.useState(null);
+  React.useEffect(() => {
+    let ignore = false;
+
+    const getEnvs = async () => {
+      if (!pipelinesBaseURI) return;
+      let appGroups: GitOpsAppGroupData[];
+      let emptyMsg = null;
+      try {
+        appGroups = await fetchAppGroups(pipelinesBaseURI, manifestURL);
+      } catch {} // eslint-disable-line no-empty
```

Comment:
I notice there is a lack of error handling. Is this acceptable? I think we might want some error handling in some fashion.

Reply:
Has a console.error message now but still added to issue [GITOPS-1278](https://issues.redhat.com/browse/GITOPS-1278) to add better error handling